spec/amp-framework-hosting.md

```diff
@@ -0,0 +1,263 @@
+
+### Meta tags
+
+If you opted to download the AMP framework, then it was built under the assumption that it would be hosted from its default location, `cdn.ampproject.org`. While `<script>` tags included in the `<head>` of your AMP pages will download from the URLs specified in their `src` attributes, dynamically loaded components like `amp-loader-0.1.js` will download from `cdn.ampproject.org`. Mixing a self-hosted AMP runtime with AMP components from `cdn.ampproject.org` is not supported and can lead to unpredictable end user experiences.
+
+The default host used by the AMP runtime can be updated by including the following `<meta>` tag in the `<head>` of your AMP pages:
+
+```
+<meta name="runtime-host" content="...">
+```
+
+where the `content` attribute should contain a URL to your hosted AMP framework. Similarly, an amp-geo fallback API URL (described in section [amp-geo hotpatching](#amp-geo-hotpatching)) can be defined with the following `<meta>` tag:
+
+```
+<meta name="amp-geo-api" content="...">
+```
+
+In each case, the value of the `content` attribute should be an absolute, HTTPS URL. Importantly, **these meta tags must appear before the first `<script>` tag**. For example, if your AMP framework is hosted at `https://example.com/amp`, then your `<head>` should look similar to the following:
+
+```
+<head>
+  ...
+  <!-- This instructs the AMP runtime to update its default host -->
+  <meta name="runtime-host" content="https://example.com/amp">
+
+  <!-- (Optional) This provides a fallback API for amp-geo -->
+  <meta name="amp-geo-api" content="https://example.com/geo/get-country">
+
+  <!-- All scripts appear after the above meta tags -->
+  <script async src="https://example.com/amp/v0.js"></script>
+  <script async custom-element="amp-geo" src="https://example.com/amp/v0/amp-geo-0.1.js"></script>
+</head>
+```
+
+## Serve the AMP framework
+
+Depending on your hosting capabilities and goals, there are several options to consider when serving the AMP framework.
+
+### URLs
+
+When the AMP runtime is initialized, all components are verified to belong to the same AMP framework version as the runtime itself. This affects components that have already downloaded as well as components that are downloaded dynamically, like amp-loader. Dynamic components and components with incorrect versions are (re-)downloaded from an rtv-specific path. This has consequences on how you choose to host the runtime.
+
+#### Versioned URLs
+
+The simplest option for hosting the AMP framework is to host it from an rtv-specific path. For example, consider AMP framework host `https://example.com/amp`. Production version `200229061636` corresponds to runtime version `01200229061636` and should be hosted from `https://example.com/amp/rtv/01200229061636`. In this case, AMP meta and script URLs would look like the following:
```

Comment:
Versions are missing the trailing 0 in this entire section

Reply:
Done